### PR TITLE
Adds tests for checking existence of Codecs in DefaultCodecs

### PR DESCRIPTION
**Pull request by gmullerb**
Based on the issue #107, the following changes are proposed:

* Adds tests for checking existence of Non Optionals, Optionals and Duplicates Codecs in `DefaultCodes`.
* Sets `client` class loader for the searching optional codecs.

### CODE REVIEW

File: src/main/java/io/r2dbc/h2/codecs/DefaultCodecs.java [hunk 1]
```diff
@@ -113,6 +92,38 @@ public Value encodeNull(Class<?> type) {
...
     }
 
+    /**
+     * Creates Default {@link Codec}s list
+     *
+     * @param client for Lobs {@link Codec}s
+     * @param classLoader to scan for classes
+     * @return a {@link List} of default {@link Codec}s
+     */
+    static List<Codec<?>> createCodecs(final Client client, final ClassLoader classLoader) {
```
[the remark this answers is not in the document]
Comment (gmullerb) [reply src/main/java/io/r2dbc/h2/codecs/DefaultCodecs.java]: No harm here (as in others methods and possible optimization from compiler), but i can remove it, please let me know.

**Comment by gmullerb** [comment]
This is the most accessible and understandable way I thought. Being a unit test, I think making fun of the class loader is reasonable, I can investigate other ways that involve an integration test.

File: src/main/java/io/r2dbc/h2/codecs/DefaultCodecs.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (katzyn) [reply src/main/java/io/r2dbc/h2/codecs/DefaultCodecs.java]: There are no possible optimizations for `final` parameters and local variables; and this modifier on them is not saved to the bytecode.

**Comment by mp911de** [comment]
Thanks for the ping. This PR got somehow lost and we want to have a look at this one.

**Comment by gmullerb** [comment]
You're Welcome.

**Comment by gmullerb** [comment]
Sure :wink:

**Comment by gmullerb** [comment]
My Pleasure!

**Pull request merged**